pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go

```diff
@@ -106,7 +106,7 @@ func (h *handler) assignWorkspace(key string, cluster *apimgmtv3.Cluster) (*apim
 		return cluster, nil
 	}
 
-	if cluster.Spec.FleetWorkspaceName == "" {
+	if cluster.Spec.FleetWorkspaceName == "" && settings.FleetManagementByDefault.Get() == "true" {
```

Comment:
```suggestion
	if cluster.Spec.FleetWorkspaceName == "" && strings.EqualFold(settings.FleetManagementByDefault.Get(), "true") {
```

Reply:
This was changed to utilize turtles label, I’d resolve it now.